### PR TITLE
Refactor analysis detail

### LINKED ISSUE #278: refactor the analysis-detail page to put the heatmap into a distinct component
This is a sub-issue (part 1) of #62.

### PR DESCRIPTION
**Pull request by mhuyck**
This PR resolves #278.

The biggest stumbling block here is teasing heatmap-specific data and logic out of the overgrown `sampleBin` service. I now have all of the heatmap data in its own service along with 2 of 6 methods migrated over from `sampleBin` that belong there.

Update: now the Sample service has its own cache of data and the methods from `sampleBin` that belong there have been moved as well.

Update 2: the heatmap is now separated into a distinct component and decoupled from the `sampleBin` service. Next steps will fully separate this new component out of the view so it can load into its own tab. In the meantime, the decoupling may look a little odd to anyone who is used to having live updates to the heatmap when editing the list of samples.

### CODE REVIEW

**Comment by mhuyck** [comment]
None of the codeclimate issues are actually new. Each one seems to be popping up anew due to code that has been moved during refactoring.

While the work done in this PR will not address the one issue I would consider valid (`rebuildHeatmapActivity()` method has too many lines), this round of refactoring will make it feasible to consider further simplification of the code in `Heatmap` when that becomes necessary.

**Comment by mhuyck** [comment]
Test server up and running at http://greenelab.fgtech.com/

**Comment by mhuyck** [comment]
I just noticed that this was still tagged "in progress" and my last update above talks about "next steps." That is unintentionally confusing... *this* PR is all done and ready for review, which I indicated by removing "WIP" from the title.

The tag is now removed to help clarify, and I'll tag @dongbohu here to be sure he sees this comment that this is ready for his feedback.

**Comment by mhuyck** [comment]
A side effect of this work is that is fixes #326.

File: interface/src/app/heatmap/heatmap-vgspec.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
+angular.module('adage.heatmap-vgspec', [])
+
+.constant('HeatmapSpec', {
```
Comment (dongbohu) [inline interface/src/app/heatmap/heatmap-vgspec.js]: :+1:

File: interface/src/app/signature/resources.js [hunk 2]
```diff
@@ -11,12 +11,6 @@ angular.module('adage.signature.resources', [
   }
 ])
 
-.factory('Activity', ['$resource', 'ApiBasePath',
```
Comment (dongbohu) [inline interface/src/app/signature/resources.js]: :+1:

**Comment by dongbohu** [comment]
By the way, I also took a look at all the new issues reported by codeclimate. I think we can safely ignore them.

File: interface/src/app/analyze/analysis/sampleBin.js [hunk 3]
```diff
@@ -8,8 +8,9 @@ angular.module('adage.analyze.sampleBin', [
   'ngResource',
   'greenelab.stats',
   'adage.utils',
-  'adage.sample.service',     // provides Sample
-  'adage.signature.resources' // provides Activity, Signature
+  'adage.sample.service',       // provides Sample
+  'adage.signature.resources',  // provides Activity, Signature
```
[the remark this answers is not in the document]
Comment (mhuyck) [reply interface/src/app/analyze/analysis/sampleBin.js]: Ooh, nice catch.

- [ ] update the comment on `adage.signature.resources`

[the remark this answers is not in the document]
Comment (mhuyck) [reply interface/src/app/analyze/analysis/sampleBin.js]: To be consistent, I also listed `adage.activity.service` as a dependency. I know it is included implicitly via `adage.heatmap.service`, but I am finding it helpful as I refactor to make all dependencies for each module fully explicit.

**Comment by mhuyck** [comment]
Thank you for the feedback, @dongbohu. I think this should be all set now, assuming CircleCI finishes happily.

**Pull request merged**